pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/typesupport/internal/PmdLanguageTypeSupport.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cli.commands.typesupport.internal;
+
+import java.util.Iterator;
+
+import net.sourceforge.pmd.lang.Language;
+import net.sourceforge.pmd.lang.LanguageRegistry;
+
+import picocli.CommandLine.ITypeConverter;
+import picocli.CommandLine.TypeConversionException;
+
+/**
+ * Provider of candidates / conversion support for supported PMD languages.
+ * 
+ * Beware, the help will report this on runtime, and be accurate to available
+ * modules in the classpath, but autocomplete will include all available at build time.
+ */
+public class PmdLanguageTypeSupport implements ITypeConverter<Language>, Iterable<String> {
+
+    @Override
+    public Language convert(final String value) throws Exception {
+        return LanguageRegistry.getLanguages().stream()
+                .filter(l -> normalizeName(l).equals(value)).findFirst()
+                .orElseThrow(() -> new TypeConversionException("Unknown language: " + value));
+    }
+
+    @Override
+    public Iterator<String> iterator() {
+        return LanguageRegistry.getLanguages().stream().map(PmdLanguageTypeSupport::normalizeName).iterator();
+    }
+    
+    public static String normalizeName(final Language lang) {
+        return lang.getTerseName().replace(' ', '-');
```

Comment:
https://github.com/pmd/pmd/pull/4060/files#diff-13b0c252d6bcd1baf3435099a4d91c5ad9fc708390836e3864aa56e75df7a706R61 strengthens the contract of getTerseName (replaced with getId) to be a java identifier. This makes the normalization redundant. All existing languages currently respect this so you could be using just getTerseName.